pandas/tests/plotting/test_groupby.py

```diff
@@ -65,3 +66,48 @@ def test_plot_kwargs(self):
 
         res = df.groupby("z").plot.scatter(x="x", y="y")
         assert len(res["a"].collections) == 1
+
+    @pytest.mark.parametrize("column, expected_axes_num", [(None, 2), ("b", 1)])
+    @pytest.mark.parametrize("label", [None, "d"])
+    def test_groupby_hist_with_legend(self, column, expected_axes_num, label):
+        # GH 6279
+        # Histogram can have a legend
+        expected_layout = (1, expected_axes_num)
+        expected_labels = label or column or [["a"], ["b"]]
+
+        index = Index(15 * ["1"] + 15 * ["2"], name="c")
+        df = DataFrame(np.random.randn(30, 2), index=index, columns=["a", "b"])
+        g = df.groupby("c")
+
+        kwargs = {"legend": True, "column": column}
+        # We get warnings if kwargs contains "label": None
+        if label is not None:
+            kwargs["label"] = label
+
+        ret = g.hist(**kwargs)
+        for (_, axes) in ret.iteritems():
```

Comment:
ret should be a list/array, lets just do `for axes in ret`, and also applies to other tests.